### PR TITLE
Store block params in data attribute

### LINKED ISSUE #494: i-bem__dom: Think about port js-params parsing from `onclick` to data-attribute
It gain dramatically performance boost, so maybe we can port in in some backward compatibility optional manner.

### PR DESCRIPTION
**Pull request by sbmaxx**
Fixes #494
/cc @dfilatov , @veged

### CODE REVIEW

File: blocks-common/i-bem/__dom/i-bem__dom.js [hunk 1]
```diff
@@ -185,13 +185,8 @@ function getParams(domNode) {
...
 function extractParams(domNode) {
 
-    var fn = domNode.onclick || domNode.ondblclick;
-    if(!fn && domNode.tagName.toLowerCase() == 'body') { // LEGO-2027 in FF onclick doesn't work on body
-        var elem = $(domNode),
-            attr = elem.attr('onclick') || elem.attr('ondblclick');
-        attr && (fn = Function(attr));
-    }
-    return fn? fn() : {};
+    var fn = domNode.getAttribute('data-bem');
```
Comment (dfilatov) [inline blocks-common/i-bem/__dom/i-bem__dom.js]: `fn` (function)? I think that is attribute value.

**Comment by dfilatov** [comment]
LGTM

**Comment by narqo** [comment]
Ouch! `touch-blocks` haven't been fixed: https://github.com/bem/bem-bl/blob/dev/blocks-touch/i-bem/__html/i-bem__html.bemhtml#L1

**Comment by sbmaxx** [comment]
@dfilatov @narqo fixed & rebased

**Comment by narqo** [comment]
I don't remember any cases in our **current** code base where something can break. But it really feels like a major changes. At least custom templates engine to work with `i-bem.js` from "our opensource-community" (PHP anyone? :) may rely on `onclick`-based behavior.

**Comment by dfilatov** [comment]
Oh. Do we support IE8? If we do -- we need to add shim for `JSON.parse`, at least as here: https://github.com/bem/es5-shims/blob/master/common.blocks/ecma/__json/ecma__json.js#L70

**Comment by sbmaxx** [comment]
@dfilatov we do support IE8. But it has native `JSON.parse` — http://caniuse.com/json

**Comment by dfilatov** [comment]
@sbmaxx IE8 has native support of `JSON.parse` only in standart mode. Most of the yandex services use compatibility mode, where `JSON.parse` is unavailable.

**Comment by sbmaxx** [comment]
@dfilatov added polyfill

**Comment by dfilatov** [comment]
@sbmaxx add `i-ecma__json` as deps for `i-bem__dom`

**Comment by sbmaxx** [comment]
@dfilatov done & rebased

**Comment by sbmaxx** [comment]
Merged.

**Pull request closed without merging**